src/credits/research.ts

```
@@ -0,0 +1,276 @@
+    }
+
+    return totals
+      .sort((a, b) => b.totalCredits - a.totalCredits)
+      .slice(0, limit);
+  }
+
+  /**
+   * Generate unique ID
+   */
+  private generateId(): string {
+    return `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
```

ESLint

[error] Make sure that using this pseudorandom number generator is safe here.

(sonarjs/pseudo-random)